### PR TITLE
[bugfix] Prevent segmentation fault for unkown keywords in parallel.

### PR DESCRIPTION
**Pull request by blattms**
If there are unknown keywords and the parser throws an exception the we nevertheless broadcasted the eclipseState and schedule. Unfortunately, these might be null pointers in this case and the serializer will run into a segmentation fault.

Broadcasting is now only done if parsing was successful.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

File: opm/simulators/utils/readDeck.cpp [hunk 1]
```diff
@@ -324,9 +324,16 @@ void readDeck(int rank, std::string& deckFilename, std::unique_ptr<Opm::Deck>& d
             eclipseState = std::make_unique<Opm::ParallelEclipseState>();
     }
 
+    // In case of parse erorors eclipseState/schedule might be null
```
Comment (bska) [inline opm/simulators/utils/readDeck.cpp:327]: Tiny nit: `erorors` should be `errors`.  Otherwise this looks good to me.

Comment (blattms) [reply opm/simulators/utils/readDeck.cpp]: Sigh, my typing just sucks. Maybe this 4.5 finger system of mine does not work :smile: 
Done and force-pushed.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
:flushed: Now?

**Comment by bska** [comment]
> 😳 Now?

🚢 !

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
Thanks for the updates.  I'll merge into master.

File: opm/simulators/utils/readDeck.cpp [hunk 2]
```diff
@@ -344,8 +352,6 @@ void readDeck(int rank, std::string& deckFilename, std::unique_ptr<Opm::Deck>& d
         errorGuard->clear();
     }
 
-
-    auto comm = Dune::MPIHelper::getCollectiveCommunication();
     parseSuccess = comm.min(parseSuccess);
```
Comment (bska) [inline opm/simulators/utils/readDeck.cpp:355]: My bad, I didn't see the implication here.  Moving the definition of `comm` into the `HAVE_MPI` section means that
```C++
parseSuccess = comm.min(parseSuccess);
```
becomes invalid in sequential builds.  We should probably protect the `comm.min()` call with `#if HAVE_MPI` too.

**Comment by blattms** [comment]
broke sequential build. Will add another fix

**Pull request merged**